### PR TITLE
Added HA support for master via zookeeper

### PR DESCRIPTION
**Pull request by amalakar**
- Added support for HA of master via zookeeper
- Added support to enable JMX for the master/worker processes for JMX monitoring
- Fixed the path for worker_runner.sh
- Removed wait for master port, since there may be multiple hosts in case of HA

### CODE REVIEW

**Comment by amalakar** [comment]
I have tested and verified HA on a test cluster with spark 1.6.1.

**Comment by amalakar** [comment]
@mbautin is this cookbook actively managed? If yes could you please take a look at the pull request. We are trying to use this cookbook to provision our spark cluster. Wanted to understand the state of this cookbook for the same reason.

**Comment by jharveysmith** [comment]
I'll take a look

File: templates/default/spark-env.sh.erb [hunk 1]
```diff
@@ -137,3 +145,35 @@ unset DEBUG_PORT FREE_DEBUG_PORT
...
+
+for arg in "$@"; do
+  case "${arg}" in
+    spark.deploy.master.Master);;
+    org.apache.spark.deploy.master.Master)
+<% if node['apache_spark']['standalone']['master_jmx_enabled'] %>
+export SPARK_DAEMON_JAVA_OPTS="$SPARK_DAEMON_JAVA_OPTS \
+  -Dcom.sun.management.jmxremote.port=<%= node['apache_spark']['standalone']['master_jmx_port'] %> \
+  -Dcom.sun.management.jmxremote.authenticate=false \
+  -Dcom.sun.management.jmxremote.ssl=false"
```
Comment (jharveysmith) [inline templates/default/spark-env.sh.erb]: Should com.sun.management.jmxremote.authenticate and com.sun.management.jmxremote.ssl be attributes too?

File: recipes/spark-standalone-worker.rb [hunk 2]
```diff
@@ -15,18 +15,25 @@
...
 include_recipe 'monit_wrapper'
 
-worker_runner_script = ::File.join(node['apache_spark']['install_dir'], 'worker_runner.sh')
+worker_runner_script = ::File.join(node['apache_spark']['install_dir'], 'bin', 'worker_runner.sh')
 worker_service_name = 'spark-standalone-worker'
 
 spark_user = node['apache_spark']['user']
 spark_group = node['apache_spark']['group']
 
+if node['apache_spark']['standalone']['master_url'] == nil
```
Comment (jharveysmith) [inline recipes/spark-standalone-worker.rb]: node['apache_spark']['standalone']['master_url'].nil?

**Comment by amalakar** [comment]
@jharveysmith thanks for the code review. I have incorporated the review comments.

**Pull request merged**